connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/CheckConnectJvmClientCompatibility.scala

```diff
@@ -185,30 +210,32 @@ class CompatibilitySuite extends ConnectFunSuite {
       .filter { p =>
         excludeRules.forall(rule => rule(p))
       }
-
-    if (problems.nonEmpty) {
-      fail(
-        s"\nComparing client jar: $clientJar\nand sql jar: $sqlJar\n" +
-          problems.map(p => p.description("client")).mkString("\n"))
-    }
+    problems
   }
 
-  test("compatibility API tests: Dataset") {
-    val clientClassLoader: URLClassLoader = new URLClassLoader(Seq(clientJar.toURI.toURL).toArray)
-    val sqlClassLoader: URLClassLoader = new URLClassLoader(Seq(sqlJar.toURI.toURL).toArray)
-
-    val clientClass = clientClassLoader.loadClass("org.apache.spark.sql.Dataset")
-    val sqlClass = sqlClassLoader.loadClass("org.apache.spark.sql.Dataset")
+  private def checkDatasetApiCompatibility(clientJar: File, sqlJar: File): Seq[String] = {
+
+    def methods(jar: File, className: String): Seq[String] = {
+      val classLoader: URLClassLoader = new URLClassLoader(Seq(jar.toURI.toURL).toArray)
```

Comment:
To be more safe about the class is loaded correctly. We can also use this class loader:
```
import org.apache.spark.util.ChildFirstURLClassLoader

val classLoader: URLClassLoader = new ChildFirstURLClassLoader(Seq(jar.toURI.toURL).toArray, this.getClass.getClassLoader)
```

Reply:
Thanks

Reply:
done

Reply:
This is effective. I have removed the change of `Dataset.scala` and it can find out `org.apache.spark.sql.Dataset.collectResult` not exist in `sql` module

Reply:
And I do manually check for the original test `compatibility API tests: Dataset`, it also became failed:

[failed.log](https://github.com/apache/spark/files/10867177/failed.log) (The error message is too large. I pasted it into the file)